---
title: "readme"
author: "Peter Searle"
date: "2024-05-14"
output: html_document
---

# Data

Description: 

data/assembly/cd-hit/transcriptomes - Final assembled transcriptomes used for phylogenetic analyses<\br>
  BearRiver.fasta.gz<\br>
  BeaverCR.fasta.gz<\br>
  BobtailCR.fasta.gz<\br>
  Bonneville1.fasta.gz<\br>
  Bonneville2.fasta.gz<\br>
  Coastal.fasta.gz<\br>
  Greenback.fasta.gz<\br>
  Humboldt.fasta.gz<\br>
  Lahontan.fasta.gz<\br>
  LWFBlackCR.fasta.gz<\br>
  RioGrande.fasta.gz<\br>
  UMCR.fasta.gz<\br>
  Westslope.fasta.gz<\br>
  Yellowstone.fasta.gz<\br>
  ncbi_files.tsv -  Information to match TSA files with original file names used in code. <\br>
  
data/phylogeny - Final concatenated and coalescent-based species trees<\br>
  17sp_13_concat.tre - Concatenated species tree generated with genes with 13 or more specimens represented<\br>
  17sp_13_coalescent.tre - Coalescent species tree generated with genes with 13 or more specimens represented<\br>
  17sp_8_concat.tre - Concatenated species tree generated with genes with 8 or more specimens represented<\br>
  17sp_8_coalescent.tre - Coalescent species tree generated with genes with 8 or more specimens represented<\br>
  17sp_4_concat.tre - Concatenated species tree generated with genes with 4 or more specimens represented<\br>
  17sp_4_coalescent.tre - Coalescent species tree generated with genes with 4 or more specimens represented<\br>

data/raw/ncbi - Information to match raw SRA files with original file names used in code.<\br>
  ncbi_files.txt - Note that the .bam files used for the transcriptome assembly are already demultiplexed because NCBI requested I provide those files, not the original movieX.subreads.bam file (which included samples multiplexed on the same run). You can run the transcriptome assembly code using IsoSeq3 starting at the "Refine" step. <\br>
  

  
  
